### PR TITLE
Add basic UI

### PR DESCRIPTION
**Pull request by NoelLincoln**
Hello 👋,

In  this milestone I was able to:

- Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:

- [ ] List all books.

- [ ] List all people.

- [ ] Create a person (teacher or student, not a plain Person).

- [ ] Create a book.

- [ ] Create a rental.

- [ ] List all rentals for a given person id.

- In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:

- [ ] Present the user with a list of options to perform.

- [ ] Lets users choose an option.

- [ ] If needed, ask for parameters for the option.

- [ ] Have a way to quit the app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  def display_menu
+    puts 'Please select an option by entering a number:'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a book'
+    puts '5. Create a rental'
+    puts '6. List all rentals for a given person id '
+    puts '7. Exit'
```
Comment (TadesseDev) [inline main.rb:34]: - [ ] Dear @NoelLincoln, You did great with adding and listing books, also adding rental seems to work fine. The main issue I face is listing all the rentals once created. The app raises an issue when I use the ID or the number of the person. Please make sure to overcome this issue and list all the rentals.

### The issue 
Screenshot 2023-11-25 at 9 35 11 at night

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @NoelLincoln 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅

 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    if person
+      puts 'Rentals:'
+      person.rentals.each do |rental|
+        puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+        puts ' '
+      end
+    else
+      puts 'Person not found.'
+    end
```
Comment (franclobo) [inline app.rb:124]: - [ ] When listing all rentals it didn't return the correct value. Please fix this issue for a better UX 👇🏽 

image

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @NoelLincoln, 

# Required Changes ♻️

istockphoto-1079018918-612x612

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Correct use of require_relative ✔️ 
- Good UI ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
```
Comment (shahadat3669) [inline app.rb:124]: 👏 Great effort in addressing the previous reviewer's comments! 👏 However, there's still an issue when listing all rentals, and it's crucial for a better user experience. Let's work on resolving this issue:

- [ ] **Fixing Issue in Listing All Rentals:** The current implementation doesn't return the correct value when listing all rentals. Please investigate and adjust your code to ensure that the listing functionality provides accurate and expected results.


https://github.com/NoelLincoln/ruby_school_library/assets/55840999/f24548f1-8cd9-4261-be82-1d1e447ab36f



By resolving this issue, you'll enhance the user experience and ensure that the listing functionality functions correctly. Keep up the good work! 🚀🔍

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @NoelLincoln, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Most of the requirement has been meet
✅ Good UI
✅ clean code 💯


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 3 above)
Comment (NoelLincoln) [reply app.rb:124]: Hello @shahadat3669 ,
Thank you. I noticed that later.

**Review by zeff96 — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW 🛑
Hi @NoelLincoln , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Hello @NoelLincoln, while you have made great effort am sorry I cannot proceed to review your project. Since you are repeating the block and your project was not approved. I would suggest you submit a new PR with all the changes required in this milestone

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**